recipes/libxml2/all/conanfile.py

```diff
@@ -230,34 +232,55 @@ def package(self):
 
-            pdb_files = glob.glob(os.path.join(self.package_folder, 'bin', '*.pdb'), recursive=True)
+            pdb_files = glob.glob(os.path.join(self.package_folder, "bin", "*.pdb"), recursive=True)
             for pdb in pdb_files:
                 os.unlink(pdb)
 
-        tools.rmdir(os.path.join(self.package_folder, 'share'))
-        tools.rmdir(os.path.join(self.package_folder, 'lib', 'cmake'))
-        tools.rmdir(os.path.join(self.package_folder, 'lib', 'pkgconfig'))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
 
     def package_info(self):
         if self._is_msvc:
-            self.cpp_info.libs = ['libxml2' if self.options.shared else 'libxml2_a']
+            self.cpp_info.components["xml2lib"].libs = ["libxml2" if self.options.shared else "libxml2_a"]
         else:
-            self.cpp_info.libs = ['xml2']
-        self.cpp_info.includedirs.append(os.path.join("include", "libxml2"))
+            self.cpp_info.components["xml2lib"].libs = ["xml2"]
+        self.cpp_info.components["xml2lib"].includedirs.append(os.path.join("include", "libxml2"))
         if not self.options.shared:
-            self.cpp_info.defines = ["LIBXML_STATIC"]
-        if self.options.include_utils:
-            bindir = os.path.join(self.package_folder, "bin")
-            self.output.info("Appending PATH environment variable: {}".format(bindir))
-            self.env_info.PATH.append(bindir)
+            self.cpp_info.components["xml2lib"].defines = ["LIBXML_STATIC"]
 
-        if self.settings.os == "Linux" or self.settings.os == "Macos":
-            self.cpp_info.system_libs.append('m')
+        if self.settings.os == "Linux":
+            self.cpp_info.components["xml2lib"].system_libs.append("m")
         if self.settings.os == "Windows":
-            self.cpp_info.system_libs.append('ws2_32')
+            self.cpp_info.components["xml2lib"].system_libs.append("ws2_32")
+        if self.options.threads:
+            if self.settings.os == "Linux":
+                self.cpp_info.components["xml2lib"].system_libs.append("pthread")
+
+        if self.options.zlib:
+            self.cpp_info.components["xml2lib"].requires.append("zlib::zlib")
+        if self.options.lzma:
+            self.cpp_info.components["xml2lib"].requires.append("xz_utils::xz_utils")
+        if self.options.iconv:
+            self.cpp_info.components["xml2lib"].requires.append("libiconv::libiconv")
+        if self.options.icu:
+            self.cpp_info.components["xml2lib"].requires.append("icu::icu")
+
+        self.cpp_info.components["xml2lib"].names["pkg_config"] = "libxml-2.0"
+        self.cpp_info.filenames["cmake_find_package"] = "LibXml2"
+        self.cpp_info.filenames["cmake_find_package_multi"] = "LibXml2"
```

Comment:
So why not `libxml2`?

Reply:
Ok, I'll remove the `cmake_find_package`.

Reply:
why removing cmake_find_package? I'm a little bit lost.

Reply:
I just checked a output and the configuration creates `libxml2-config.cmake`.
So, as you said, it should set the `cmake_find_package_multi` name to `libxml2`.
So I think `cmake_find_package` should not be set.
Or should it be set to a *hidden* name? Like `_libxml2`.
Or should it be set to the same name as `cmake_find_package_multi`?

Reply:
`LibXml2` for `cmake_find_package` is fine: https://cmake.org/cmake/help/latest/module/FindLibXml2.html

But all of this (module and config) doesn't require components (except if you anticipate "executable components")